### PR TITLE
chore(core): enable wandb-core for all the functional tests

### PR DESCRIPTION
**Pull request by kptkin**
Description
-----------

- Fixes WB-17502

What does the PR do?

- Turn on `wandb-core` for all the functional tests and mark the failing test with a `wandb_core_failure` shard.
- rename gotests to unit-tests-go
- move all functional integration tests to nightly and create a shard based on integration
- create shards for sdk, artifact, launch

Testing
-------
How was this PR tested?

This is a test PR

### CODE REVIEW

**Review by timoffex — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -1278,6 +1280,50 @@ workflows:
           requires:
             - "slack-notify-on-start"
 
+      - tox-base:
+          matrix:
+            parameters:
+              python_version_major: [3]
+              python_version_minor: [7]
+              with_core: [true]
+              shard: ["tf115", "tf21", "tf24", "tf25", "tf26", "xgboost"]
```
[the remark this answers is not in the document]
Comment (kptkin) [reply .circleci/config.yml:1289]: good point, i was just copy-pasting it around

File: tests/functional_tests/t0_main/profiler/profiler.yea [hunk 2]
```diff
@@ -2,6 +2,7 @@ id: 0.profiler
 plugin:
   - wandb
 tag:
+  shard: torch
```
[the remark this answers is not in the document]
Comment (kptkin) [reply tests/functional_tests/t0_main/profiler/profiler.yea:5]: oh it is a profiler but comes from torch so i just used this shard i can change it to profiler

File: .circleci/config.yml [hunk 3]
```diff
@@ -1278,6 +1280,50 @@ workflows:
...
+                - "fastai"
+                - "catboost"
+                - "jax"
+                - "prefect"
+                - "prodigy"
+                - "torch"
+                - "sacred"
+                - "sb3"
+                - "tf"
+                - "lightning"
```
[the remark this answers is not in the document]
Comment (kptkin) [reply .circleci/config.yml:1319]: again good point, just copy pasted as well. will fix.

File: .circleci/config.yml [hunk 4]
```diff
@@ -481,6 +481,9 @@ jobs:
       tox_args:
         type: string
         default: ""
+      with_core:
+        type: boolean
+        default: false
```
[the remark this answers is not in the document]
Comment (kptkin) [reply .circleci/config.yml:486]: will check

File: tests/functional_tests/t0_main/save/save-dir.py [hunk 5]
```diff
@@ -4,6 +4,11 @@
 id: 0.save.dir
 plugin:
   - wandb
+tag:
+  skip: true
```
[the remark this answers is not in the document]
Comment (kptkin) [reply tests/functional_tests/t0_main/save/save-dir.py:8]: i added the other shard just as a marker so we can find all the tests that are failing... (it's not as nice as pytest where we can use the marker for collections)

File: .circleci/config.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kptkin) [reply .circleci/config.yml:486]: used in functional tests

**Pull request merged**